### PR TITLE
Updates for cloud autort and blstores

### PR DESCRIPTION
**Pull request by zach1221**
## PR Author Checklist:

- [x] I have linked PR's from all sub-components involved in section below. 
- [x] I am confirming reviews are completed in ALL sub-component PR's.
- [x] I have run the full RT suite on either Hera/Cheyenne AND have attached the log to this PR below this line:
  - LOG: 
- [ ] I have added the list of all failed regression tests to "Anticipated changes" section.
- [x] I have filled out all sections of the template.

## Description

Adding new blstore directories to bl.py and rt.sh so code management to transfer new baselines as needed. Adding separate noaacloud jenkins file. Updating jenkins-ci pipeline jenkins file to include sonarqube. Added noaacloud machine name to rt.conf to stipulate which tests should be run on the cloud platforms.

### Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [ ] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [x] none

### Input data
- [x] No changes are expected to input data.
- [ ] Changes are expected to input data:
  - [ ] New input data.
  - [ ] Updated input data.

### Regression Tests:
- [x] No changes are expected to any regression test.
- [ ] Changes are expected to the following tests:

<details><summary>Tests effected by changes in this PR:</summary>

</details>

### Libraries

- [x] Not Needed
- [ ] Needed
  - [ ] Create separate issue in [JCSDA/spack-stack](https://github.com/JCSDA/spack-stack) asking for update to library. Include library name, library version.
  - [ ] Add issue link from JCSDA/spack-stack following this item 

<details><summary>Code Managers Log</summary>

- [ ] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.
  - [ ] N/A

### Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Jet
  - [x] Gaea
  - [ ] Cheyenne
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- CI
  - [ ] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment
</details>

### CODE REVIEW

File: tests/rt.sh [hunk 1]
```diff
@@ -563,8 +563,11 @@ echo "Start Regression test" >> ${REGRESSIONTEST_LOG}
 echo                         >> ${REGRESSIONTEST_LOG}
 echo "Testing UFSWM Hash:" `git rev-parse HEAD` >> ${REGRESSIONTEST_LOG}
 echo "Testing With Submodule Hashes:" >> ${REGRESSIONTEST_LOG}
+cd ..
```
Comment (BrianCurtis-NOAA) [inline tests/rt.sh:566]: I know the error well. I think newer git installs do OK without this change needed. I'm OK with this change if you need the backwards compatibility, though.

**Comment by DeniseWorthen** [comment]
@zach1221 What criteria was used for deciding which tests would be enabled on noaacloud? You have for example 4 HAFS ATM tests and no CDEPS DATM tests. There are no global coupled tests except for the PDLIB (unstructure waves). How were these tests decided?

**Comment by zach1221** [comment]
> @zach1221 What criteria was used for deciding which tests would be enabled on noaacloud? You have for example 4 HAFS ATM tests and no CDEPS DATM tests. There are no global coupled tests except for the PDLIB (unstructure waves). How were these tests decided?

@DeniseWorthen I tried to keep the number of tests low due to the cost of running them on the cloud, and attempted to spread them out. I can add a few CDEPS DATM and couple more global tests though.

File: tests/rt.conf [hunk 2]
```diff
@@ -116,194 +116,194 @@ RUN | regional_ifi_2threads                             | + acorn
...
-RUN | hafs_regional_storm_following_1nest_atm_ocn_debug | - jet s4                   | baseline |
-
-COMPILE | hafsw_faster | intel | -DAPP=HAFSW -DMOVING_NEST=ON -DCCPP_SUITES=FV3_HAFS_v1_gfdlmp_tedmf_nonsst -D32BIT=ON -DFASTER=ON | -jet s4 | fv3 |
-RUN | hafs_regional_storm_following_1nest_atm_ocn_wav   | - jet s4                   | baseline |
-
-COMPILE | hafs_all | intel | -DAPP=HAFS-ALL -DCCPP_SUITES=FV3_HAFS_v1_gfdlmp_tedmf,FV3_HAFS_v1_gfdlmp_tedmf_nonsst -D32BIT=ON | - wcoss2 | fv3 |
-RUN | hafs_regional_docn                                | - wcoss2                   | baseline |
-RUN | hafs_regional_docn_oisst                          | - wcoss2                   | baseline |
-RUN | hafs_regional_datm_cdeps                          | - wcoss2 -jet s4           | baseline |
+RUN | hafs_regional_atm_ocn_wav                         | - noaacloud                | baseline |
```
Comment (DeniseWorthen) [inline tests/rt.conf]: Could you please re-align the baseline columns where needed?

**Comment by BrianCurtis-NOAA** [comment]
wcoss2: 
```
Found Python: /usr/bin/python3.6
Calling CCPP code generator (ccpp_prebuild.py) for suites --suites=FV3_HAFS_v1_.gfdlmp_tedmf_nonsst ...
CMake Error at FV3/ccpp/CMakeLists.txt:40 (message):
  An error occured while running ccpp_prebuild.py, check
  /lfs/h2/emc/ptmp/brian.curtis/FV3_RT/rt_85932/compile_hafsw_faster_intel/build_fv3_hafsw_faster_intel/F
V3/ccpp/ccpp_prebuild.{out,err}
```
location: /lfs/h2/emc/ptmp/brian.curtis/FV3_RT/rt_85932/compile_hafsw_faster_intel

**Comment by zach1221** [comment]
> wcoss2:
> 
> ```
> Found Python: /usr/bin/python3.6
> Calling CCPP code generator (ccpp_prebuild.py) for suites --suites=FV3_HAFS_v1_.gfdlmp_tedmf_nonsst ...
> CMake Error at FV3/ccpp/CMakeLists.txt:40 (message):
>   An error occured while running ccpp_prebuild.py, check
>   /lfs/h2/emc/ptmp/brian.curtis/FV3_RT/rt_85932/compile_hafsw_faster_intel/build_fv3_hafsw_faster_intel/F
> V3/ccpp/ccpp_prebuild.{out,err}
> ```
> 
> location: /lfs/h2/emc/ptmp/brian.curtis/FV3_RT/rt_85932/compile_hafsw_faster_intel

Do you think a change in my PR caused this? I haven't starting the RTs yet as jenkins-ci is still running, but I'll let you know if I have issues with this build.

File: tests/rt.conf [hunk 3]
```diff
@@ -116,194 +116,194 @@ RUN | regional_ifi_2threads                             | + acorn
...
+RUN | hafs_regional_specified_moving_1nest_atm          | - jet s4  noaacloud           | baseline |
+RUN | hafs_regional_storm_following_1nest_atm           | - jet s4  noaacloud           | baseline |
+RUN | hafs_regional_storm_following_1nest_atm_qr        | - jet s4 cheyenne noaacloud   |          |
+RUN | hafs_regional_storm_following_1nest_atm_ocn       | - jet s4  noaacloud           | baseline |
+RUN | hafs_global_storm_following_1nest_atm             | - jet s4  noaacloud           | baseline |
+
+COMPILE | hafsw_debug | intel | -DAPP=HAFSW -DMOVING_NEST=ON -DCCPP_SUITES=FV3_HAFS_v1_gfdlmp_tedmf_nonsst -D32BIT=ON -DDEBUG=ON | - jet noaacloud s4 | fv3 |
+RUN | hafs_regional_storm_following_1nest_atm_ocn_debug | - jet s4  noaacloud           | baseline |
+
+COMPILE | hafsw_faster | intel | -DAPP=HAFSW -DMOVING_NEST=ON -DCCPP_SUITES=FV3_HAFS_v1_.gfdlmp_tedmf_nonsst -D32BIT=ON -DFASTER=ON | -jet noaacloud  s4 | fv3 |
```
Comment (BrianCurtis-NOAA) [inline tests/rt.conf]: There's a typo here, remove the `.` before gfdlmp

**Review by BrianCurtis-NOAA — CHANGES_REQUESTED** [review]

File: tests/rt.conf (hunk 3 above)
Comment (zach1221) [reply tests/rt.conf]: Done

File: tests/rt.conf (hunk 2 above)
Comment (zach1221) [reply tests/rt.conf]: Columns should be straightened out now.

**Comment by FernandoAndrade-NOAA** [comment]
[ufs-weather-model » ort-docker-pipeline » PR-1887#1 Console [Jenkins].txt](https://github.com/ufs-community/ufs-weather-model/files/12629412/ufs-weather-model.ort-docker-pipeline.PR-1887.1.Console.Jenkins.txt)
Jenkins ci log attached, ORTs passed.

**Comment by zach1221** [comment]
@jkbk2004 @FernandoAndrade-NOAA I appears the Cheyenne account I've been using to run tests is now overspent and the queue disabled. So I think we'll be forced to skip Cheyenne.

**Comment by jkbk2004** [comment]
I agree we need to move to Derecho.

**Comment by jkbk2004** [comment]
@zach1221 resolve the conversations.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by sadeghitabas — APPROVED** [review]

**Pull request merged**